### PR TITLE
MENT-425 Implement a method to add IPs to allowlist for Galera Cluste…

### PR DESCRIPTION
**Pull request by mkaruza**
…r node addresses that can make SST/IST requests

### CODE REVIEW

**Review by sciascid — COMMENTED** [review]
This is a request for clarification: the whitelist table is replicated, right?
And if so, is it possible that the contents of the whitelist table may diverge on different nodes, if nodes have configured their wsrep_whitelist variable differently?

So for example, consider this scenario:
Inittially you start one node1, which has ip-x in wsrep_whitelist variable. Then node2 joins, and it has ip-x and ip-y, in its wsrep_whitelist variable.
What is the contents of the whitelist table in the two nodes?

**Comment by mkaruza** [comment]
@sciascid you are right about this, on node2 we would receive all information from donor node, but we are going to append missing allowlist entries from `wsrep_allowlist` without replication. I think approach for this is that *only* node which bootstrapped cluster should set initial allowlist. Comments about this ?

File: sql/wsrep_mysqld.cc [hunk 1]
```diff
@@ -1140,6 +1158,16 @@ bool wsrep_start_replication(const char *wsrep_cluster_address)
 
   DBUG_ASSERT(wsrep_cluster_address[0]);
 
+  // --wsrep-new-cluster flag is not used, checking wsrep_cluster_address
+  // it should match gcomm:// only to be considered as bootstrap node.
+  // This logic is used in galera.
+  if (!wsrep_new_cluster && 
+      (strlen(wsrep_cluster_address) == 8) &&
```
Comment (sciascid) [inline sql/wsrep_mysqld.cc:1165]: This call to `strlen()` seems unnecessary. 
Consider removing it.

**Review by sciascid — CHANGES_REQUESTED** [review]
Add a test that shows only bootstrap server adds entries from allowlist variable, while other nodes ignore the variable.

File: sql/wsrep_mysqld.cc (hunk 1 above)
Comment (sciascid) [reply sql/wsrep_mysqld.cc:1165]: Disregard my comment above.

**Review by sciascid — APPROVED** [review]
Looks good now

**Comment by mkaruza** [comment]
Closed and opened against MariaDB

**Pull request closed without merging**